Project.toml

```diff
@@ -28,6 +28,7 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 SparseDiffTools = "47a9eef4-7e08-11e9-0b38-333d64bd3804"
 StaticArraysCore = "1e83bf80-4336-4d27-bf5d-d5a4f845583c"
 SymbolicIndexingInterface = "2efcf032-c050-4f8e-a9bb-153293bab1f5"
+TaylorDiff = "b36ab563-344f-407b-a36a-4f200bebf99c"
```

Comment:
TaylorDiff brings in Symbolics, which seems like quite a heavy dep. We have quite a few downstream users notably OrdinaryDiffEq.

Won't this affect load times? @ChrisRackauckas @oscardssmith thoughts on this? Should we make it a weak dep

Reply:
Can we make this be an extension alg in a lib like what we're doing in OrdinaryDiffEq.jl? NonlinearSolve.jl is going to have the same issue of just having too many algorithms at some point, and this seems like something that can be a lib extension